zorg/buildbot/builders/UnifiedTreeBuilder.py

```diff
@@ -567,3 +571,448 @@ def getCmakeWithNinjaMultistageBuildFactory(
            **kwargs)
 
     return f
+
+
+def getCmakeExBuildFactory(
+        depends_on_projects = None,
+        enable_runtimes = "auto",
+        cmake_definitions = None,
+        cmake_options = None,
+        allow_cmake_defaults = True,    # Add default CMake definitions to build LLVM project (if not specified).
+        targets = ["."],
```

Comment:
Please do not use mutable objects as defaults. This leads to hard-to-catch bugs. Instead, set it to `None` and handle that in the code by setting up the default value if `None`.

Reply:
Sorry, I need a usage of these default values to support a intended logic. I don't see any reasons to do not use them since I don't change them within the function. This is an useful feature of Python.

Reply:
What's wrong with
```
targets = None,
...
if targets is None:
    targets = ["."]
...
```
?
Empty list would mean no targets to build.

Reply:
In regular meaning a passing of None supposes doing nothing; or don't use this argument. A passing of something means doing something with this parameter value, even if it's an empty list (probably some default behaviour). This is an optimal meaning I think. Sure, we can change it and a passing of None will mean "use default value", but a passing of an empty list will mean "do nothing". But actually I don't see a good reason for it.

Maybe I could remove all default targets for all build operations such as check-all & etc. and force people to explicitly specify the targets everywhere to allow the step, but I would prefer to enable the default build by default and let people explicitly disable this step if necessary (by passing None).
.

Reply:
Make it immutable then. Tuple, for example
```
targets = ("."),
```
would do what you want without following a python anti-pattern.

Reply:
I used the strings instead of the lists there.